app/statistics/[cycle]/page.tsx

```diff
@@ -0,0 +1,52 @@
+import AdmissionsCycleStatistics from '@/components/AdmissionsCycleStatistics'
+import prisma from '@/db'
+import { Decision, NextAction } from '@prisma/client'
+
+export const dynamic = 'force-dynamic'
+
+export default async function AdmissionsCycleStatisticsPage({
+  params
+}: {
+  params: { cycle: string }
+}) {
+  const cycle = parseInt(params.cycle)
+
+  const outcomes = await prisma.outcome.findMany({
+    where: {
+      application: {
+        admissionsCycle: cycle
+      }
+    }
+  })
+
+  const noApplications = outcomes.length
+  const noOffers = outcomes.filter((outcome) => outcome.decision === Decision.OFFER).length
+  const noRejections = outcomes.filter((outcome) => outcome.decision === Decision.REJECT).length
+
+  const nextActionCountsQuery = await prisma.application.groupBy({
+    by: ['nextAction'],
+    where: {
+      admissionsCycle: cycle
+    },
+    _count: {
+      nextAction: true
+    }
+  })
+
+  const nextActionCounts: { name: string; quantity: number }[] = []
+  for (const nextAction of Object.keys(NextAction)) {
+    const correspondingNumber = nextActionCountsQuery.find((item) => item.nextAction === nextAction)
+      ?._count.nextAction
+    nextActionCounts.push({ name: nextAction, quantity: correspondingNumber ?? 0 })
+  }
```

Comment:
How about leveraging on functional operators here?

```suggestion
  const nextActionCounts: { name: string; quantity: number }[] = Object.keys(NextAction).map((na) => 
    ({ name: na, quantity: nextActionCountsQuery.find((i) => i.nextAction === na)?._count.nextAction ?? 0 })
  )
```
Note I am using indeed naming shortcuts here like `i` and `na`, which may seem contrary to my prev comment. With the functional operators, the reading flows naturally left to right, so each item of the map is an element in the collection you are mapping on e.g. in sequence, you have `keys(NextAction)` and then `map(na)`. This is just a personal opinion to find convenient compromises favouring conciseness where the meaning of the variables is less ambiguous. What do you think?

Reply:
The functional approach is nicer here I agree, replaced in 835db30627fd8f66c9d3e5ff6fc9086b644b2239